### ISSUE TITLE
Add support for route handling without requiring a `Component`

### ISSUE THREAD

**Issue by HermanBovens** [vaadin/flow#10419]
At the moment, in order to handle navigation to a particular route by forwarding to one of several views depending on a certain condition, one needs to create a dummy `Component` and either annotate that with `@Route`, or use `RouteConfiguration.setRoute(...)` which also requires a `Class<? extends Component>`.  Since there is no actual view that should be shown for this route, it shouldn't be necessary to create a dummy component class which will also get instantiated at runtime. 

Consider support for registering a route with custom navigation handling that doesn't require a UI component.

**Comment by pleku** [comment]
Hi. I'd consider this a duplicate of #2666 even though you're not explicitly mentioning using a method here.

**Comment by HermanBovens** [comment]
Not really, since the method must also return a Component.  The thing is,
sometimes you want to handle a route always by redirecting/forwarding, so
there is no Component.  It should not be necessary to create a dummy
Component that is never attached.

Op do 8 apr. 2021 15:26 schreef Pekka Hyvönen ***@***.***>:

**Comment by HermanBovens** [comment]
@pleku Will you re-open and remove the duplicate tag?  This question is not about using a method to define the route's target component.  @Legioth has provided good arguments in #2666 against that.  This question is rather about somehow being able to handle navigation to a certain path (not linked to any `Component`) by forwarding/re-routing to a real route (which is actually linked to a component class).

**Comment by pleku** [comment]
Ok @HermanBovens - can you please provide more details on the use case you have and how this solves that ?

I would just use a dummy component for the registered route and then either forward to the correct route component to be showed or just add the correct component as the child component.

I'm speculating, but I guess what you'd want to do (?) is register some _handler_ that is invoked for a certain route path and then it forwards to another route instead.  Like for example a class with `Route` and implements a one method interface where it gets routing data and a callback to call to make the forward/redirect - which will give the router the component(s) that should be actually shown. And in case the callback is not invoked, then the 404 error page would be shown.

**Comment by HermanBovens** [comment]
That is indeed my use case: handling A URL and, depending on user's
permissions, forward/redirect to a specific route.
Using a dummy Component works, but feels like a workaround.

Op ma 10 mei 2021 08:25 schreef Pekka Hyvönen ***@***.***>:

**Comment by HermanBovens** [comment]
@Legioth That seems correct.  But I think the term _route_ is applicable in any case, whether it is handled by an intermediate action that delegates or by a terminal action that directly renders a component.  On the other hand, for this case it doesn't seem necessary to be able to programmatically navigate to this kind of route (other than by setting window.location to the corresponding URL).

**Comment by Legioth** [comment]
I see _route_ as the URL pattern and the _action_  as the thing that happens when navigating to a URL that matches the pattern. Maybe that distinction is redundant and everything can just be referred to as _route_?

**Comment by HermanBovens** [comment]
I think the terms _route_ and _route handler_ would be appropriate and easy
to understand, but for the most used case where the route is handled by
rendering a component, the route handling should be implicit, as it is now
with the annotation.

Op wo 12 mei 2021 14:39 schreef Leif Åstrand ***@***.***>:
